nomenclature/core.py

```diff
@@ -8,11 +10,49 @@
 
 @validate_arguments(config=dict(arbitrary_types_allowed=True))
 def process(
-    df: IamDataFrame,
+    df: pyam.IamDataFrame,
     dsd: DataStructureDefinition,
-    dimensions: list = None,
-    processor: RegionProcessor = None,
-):
+    dimensions: Optional[List[str]] = None,
+    processor: Optional[RegionProcessor] = None,
+) -> pyam.IamDataFrame:
+    """Function for validation and region aggregation in one step
+
+    Since there are some subtleties in the order of applying region processing and
+    validation, this function is the recommended way of using the nomenclature package
+    as it takes care of subtleties.
+
+    Parameters
+    ----------
+    df : pyam.IamDataFrame
+        Input data to be validated and aggregated.
+    dsd : DataStructureDefinition
+        Data templates that are used for validation.
+    dimensions : Optional[List[str]], optional
+        List of dimensions that will be used in the validation, if None all
+        dsd.dimensions will be used, by default None
+    processor : Optional[RegionProcessor], optional
+        Region processor that will perform region renaming and aggregation if provided,
+        by default None
```

Comment:
The "by default None" can be seen from the function signature no need to repeat
```suggestion
        Region processor that will perform region renaming and aggregation if provided
```